stumpy/motifs.py

```diff
@@ -466,18 +484,23 @@ def match(
     if np.any(np.isnan(Q)) or np.any(np.isinf(Q)):  # pragma: no cover
         raise ValueError("Q contains illegal values (NaN or inf)")
 
-    if len(Q.shape) == 1:
+    m = Q.shape[-1]
+    excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+
+    Q_subseq_isconstant = core.rolling_isconstant(Q, m, Q_subseq_isconstant)
+    T_subseq_isconstant = core.rolling_isconstant(T, m, T_subseq_isconstant)
+
+    if Q.ndim == 1:
         Q = Q[np.newaxis, :]
-    if len(T.shape) == 1:
-        T = T[np.newaxis, :]
+    if Q_subseq_isconstant.ndim == 1:
```

Comment:
This could work?

```
if Q.ndim == 1 and Q.ndim == Q_subseq_isconstant.ndim:
    Q = Q[np.newais, :] Q_subseq_isconstant = 
    Q_subseq_isconstant[np.newaxis, :]
```

Reply:
But the multiple if statements isn't horrible

Reply:
I think the check `Q.ndim == Q_subseq_isconstant.ndim` may not be effecitve here. Note that we have
```
Q_subseq_isconstant = core.rolling_isconstant(Q, m, Q_subseq_isconstant)
```
before this if-block. Hence, I assume Q_subseq_isconstant and Q must be of same ndim, and if they are not, an error (probably?) will be raised in `core.rolling_isconstant`.

Reply:
Sounds good.